File: Phase-5/backend/src/api/schemas/notification.py
"""Notification response schemas."""
import uuid
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel


class NotificationRead(BaseModel):
    id: uuid.UUID
    user_id: str
    task_id: Optional[uuid.UUID] = None
    message: str
    read: bool
    created_at: datetime

    class Config:
        from_attributes = True


class NotificationListResponse(BaseModel):
    notifications: List[NotificationRead]
    total_count: int
    unread_count: int

File: Phase-5/frontend/src/types/notification.ts
export interface NotificationRead {
  id: string;
  user_id: string;
  task_id: string | null;
  message: string;
  read: boolean;
  created_at: string;
}

export interface NotificationListResponse {
  notifications: NotificationRead[];
  total_count: number;
  unread_count: number;
}
